### PR TITLE
Histogram Equalization

### PR DESCRIPTION
**Pull request by JoeCx**
# Overview

**Type of Change:** New Feature

**Summary:** This branch implements a new transformation class, HistogramEqualization, which transforms a tensor image using histogram equalization to enhance the contrast of the image. This transformation is then used for the dorsal images, as that is what we are currently refining. Furthermore, I added to the simulator file, so that you can see the number of images for each angle we have.

## UI/UX Implementation

No changes made.

## Model Updates

This section describes changes that were made to the models used by your application.

### Data Models

No changes made.

### Object-Oriented Models

Transformation class, HistogramEqualization, added to Training Program file. This class can be inputted to any transforms.compose call.

## End-to-End Testing Instructions

Run simulator file to test and see how updated dorsal model works.

### CODE REVIEW

File: src/simulator.py [hunk 1]
```diff
@@ -21,6 +21,13 @@
     dbr = DatabaseReader("training.db")
     df = dbr.get_dataframe()
 
+    # Display how many images we have for each angle
+    print("Number of Images for Each Angle:")
+    print(f"CAUD: {(df['View'] == 'CAUD').sum()}")
+    print(f"DORS: {(df['View'] == 'DORS').sum()}")
+    print(f"FRON: {(df['View'] == 'FRON').sum()}")
+    print(f"LATE: {(df['View'] == 'LATE').sum()}")
```
Comment (tmckeown311) [inline src/simulator.py:29]: Very helpful for debugging and ensuring proper results from database conversion

File: src/training_program.py [hunk 2]
```diff
@@ -266,7 +266,7 @@ def training_evaluation_lateral(self, num_epochs, train_loader, test_loader):
 
                 running_loss += loss.item()
 
-                print(f"Epoch {epoch+1}/{num_epochs}, Loss: {running_loss/len(train_loader):.4f}")
+            print(f"Epoch {epoch+1}/{num_epochs}, Loss: {running_loss/len(train_loader):.4f}")
```
Comment (tmckeown311) [inline src/training_program.py:269]: good bug fix

File: src/training_program.py [hunk 3]
```diff
@@ -576,3 +576,26 @@ def __call__(self, tensor):
         edge_magnitude = edge_magnitude / edge_magnitude.max()
 
         return edge_magnitude.squeeze(0)
+
+class HistogramEqualization():
```
Comment (tmckeown311) [inline src/training_program.py:580]: Good function. Maintains simplicity and uses tools made to go with the model being used which has already proven better results with the training tests we've done.

**Review by tmckeown311 — APPROVED** [review]
Changes made provide more debugging accuracy and histogram idea has proven to be very effective with model accuracy based on trial runs so far.
